### DISCUSSION TITLE
How to Publish Trimmed or AOT on Windows?

### DISCUSSION THREAD

**Discussion by klimatr26** [dotnet/maui#26031] [Q&A] [answered]
Hello.

I have been trying to publish a MAUI app as NativeAOT. I have created a new project on Visual Studio and have also used `dotnet new maui`. In both cases, `dotnet publish -f net9.0-windows10.0.19041.0 -c Release` produces an executable (and may other files) that takes a second to load but works. However, when I add `<PublishAot>true</PublishAot>` I get these messages (I am using .NET 9):
```
    C:\Users\User\.nuget\packages\microsoft.web.webview2\1.0.2792.45\lib_manual\net6.0-windows10.0.17763.0\Microsoft.Web.WebView2.Core.Projection.dll : warning IL2104: Assembly 'Microsoft.Web.WebView2.Core.Projection' produced trim warnings. For more information see https://aka.ms/il2104
    C:\Users\User\.nuget\packages\microsoft.web.webview2\1.0.2792.45\lib_manual\net6.0-windows10.0.17763.0\Microsoft.Web.WebView2.Core.Projection.dll : warning IL3053: Assembly 'Microsoft.Web.WebView2.Core.Projection' produced AOT analysis warnings.
    ILC : Trim analysis warning IL2026: WinRT.BaseActivationFactory._ActivateInstance<I>(): Using member 'WinRT.ComWrappersSupport.GetObjectReferenceForInterface<I>(IntPtr)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. This method is not trim-safe, and is only supported for use when not using trimming (or AOT).
    ILC : Trim analysis warning IL2026: WinRT.WinrtModule.GetActivationFactory(String): Using member 'WinRT.ComWrappersSupport.GetObjectReferenceForInterface<IActivationFactoryVftbl>(IntPtr)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. This method is not trim-safe, and is only supported for use when not using trimming (or AOT).
    ILC : Trim analysis warning IL2026: WinRT.DllModule.GetActivationFactory(String): Using member 'WinRT.ComWrappersSupport.GetObjectReferenceForInterface<IActivationFactoryVftbl>(IntPtr)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. This method is not trim-safe, and is only supported for use when not using trimming (or AOT).
```
When I launch the executable, nothing seems to happen. Updating WebView2 seems to solve the first two issues, but not the other three. Event Viewer says that an Application Error occurred, on module `Microsoft.UI.Xaml.dll`, with exception code `0xc000027b`.
The same thing has happened on my Windows 11 laptop and my Windows 10 desktop. Adding `<WindowsSdkPackageVersion>10.0.19041.56</WindowsSdkPackageVersion>` and `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` also makes no difference.

Edit: `<PublishTrimmed>true</PublishTrimmed>` also breaks the program.
Publishing a WinUI 3 app as Native AOT seems to work.

What can I do?
Thank you.

**Comment by ganyl** [comment]
I'm getting the same error and I debug and find the problem.
![捕获](https://github.com/user-attachments/assets/7492b1ce-6eef-465f-aa6a-27f25cabdab7)

**Comment by Piotr-Kniaz** [comment]
I encountered the same problem. May be issue with Windows App SDK? Has anyone already solved this issue?

Is NativeAOT fully supported in .NET MAUI for Windows?

**Comment by ganyl** [comment]
这是来自QQ邮箱的自动回复邮件，您的邮件我已接收，如有紧急情况请电话联系，谢谢。

**Comment by klimatr26** [comment, marked as answer]
I have just installed the latest VS and .NET 10 and created a new .NET MAUI project, and then tried to publish. I no longer get any trimmer warnings and the program starts and works properly. Sample content still produces warnings and an executable that does not start, however.

**Comment by ganyl** [comment]
这是来自QQ邮箱的自动回复邮件，您的邮件我已接收，如有紧急情况请电话联系，谢谢。
